### PR TITLE
add fr and nl translations

### PR DESCRIPTION
**Pull request by seavor**

### CODE REVIEW

**Comment by ebbit1q** [comment]
on beta the password field is untranslated also the country dropdown when registering is completely untranslated why do we have every name of the country in english?

**Comment by ebbit1q** [comment]
correction, the password field is only translated when you reload the page

**Comment by seavor** [comment]
> the country dropdown when registering is completely untranslated why do we have every name of the country in english?

We have a follow up task to translate the country dropdown.

**Comment by ebbit1q** [comment]
do we really want to translate every country name into every language?

**Comment by seavor** [comment]
> do we really want to translate every country name into every language?

Part of the hold up is i think some discussion is needed about what exactly we what to do there.

File: webclient/src/types/languages.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import { CountryLabel } from './countries';
 
 export enum Language {
-  'en-US' = 'en-US',
-  'es-ES' = 'es-ES',
+  'en' = 'en',
```
Comment (ebbit1q) [inline webclient/src/types/languages.ts:4]: the client currently uses en-US but fr and de for most languages, we also have pt-BR and zh-hans as examples, yet on the client this is all internal, I recommend adding what the language calls itself as a translatable tag

**Review by ebbit1q — DISMISSED** [review]
works on beta

File: webclient/src/types/languages.ts (hunk 1 above)
Comment (seavor) [reply webclient/src/types/languages.ts:4]: region-based translations have been another TBD discussion item, lets make it a follow up.

File: webclient/src/components/LanguageDropdown/LanguageDropdown.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ const LanguageDropdown = () => {
             return (
               <MenuItem value={lang} key={lang}>
                 <div className="LanguageDropdown-item">
-                  <img className="LanguageDropdown-item__image" src={Images.Countries[country]} alt={CountryLabel[country]} />
```
Comment (seavor) [inline webclient/src/components/LanguageDropdown/LanguageDropdown.tsx:39]: alt text intentionally removed as it rendered on top of the label.

File: webclient/src/common.i18n.json [hunk 3]
```diff
@@ -22,6 +22,258 @@
...
+      "CN": "China",
+      "CX": "Christmas Island",
+      "CC": "Cocos (Keeling) Islands",
+      "CO": "Colombia",
+      "KM": "Comoros",
+      "CG": "Congo",
+      "CD": "The Democratic Republic of the Congo",
+      "CK": "Cook Islands",
+      "CR": "Costa Rica",
+      "CI": "Côte d\"Ivoire",
```
Comment (ebbit1q) [inline webclient/src/common.i18n.json:81]: why \\" ? also why is ivory coast not translated?

File: webclient/src/components/CountryDropdown/CountryDropdown.tsx [hunk 4]
```diff
@@ -3,15 +3,28 @@ import React, { useEffect, useState } from 'react';
...
 
 const CountryDropdown = ({ input: { onChange } }) => {
   const [state, setState] = useState('');
+  const [sortedCountries, setSortedCountries] = useState([]);
+  const { t, i18n } = useTranslation();
 
   useEffect(() => onChange(state), [state]);
 
+  useEffect(() => {
+    const collator = new Intl.Collator(i18n.language);
```
Comment (seavor) [inline webclient/src/components/CountryDropdown/CountryDropdown.tsx:21]: https://www.measurethat.net/Benchmarks/Show/12644/1/intlcollatorcompare-vs-localecompare

File: webclient/src/common.i18n.json (hunk 3 above)
Comment (ebbit1q) [reply webclient/src/common.i18n.json]: hey, if you're renaming that one what about fixing saint helena? what about north macedonia? how many countries recognize kosovo (XK)?

**Review by ebbit1q — APPROVED** [review]
I guess fixing up the contents of the country list is out of scope, good job on the sorting, looks good even though the menu is ever so sluggish

**Comment by seavor** [comment]
> I guess fixing up the contents of the country list is out of scope, good job on the sorting, looks good even though the menu is ever so sluggish

I created tickets for both of these issues

**Pull request merged**